sql/driver/driver_it_test.go

```diff
@@ -527,6 +528,46 @@ func TestContextCancelAfterFirstPage(t *testing.T) {
 	})
 }
 
+func TestClusterShutdownDuringQuery(t *testing.T) {
+	it.SkipIf(t, "hz < 5.0")
+	ctx := context.Background()
+	port := it.NextPort()
+	cls := it.StartNewClusterWithConfig(it.MemberCount(), it.SQLXMLConfig(t.Name(), "localhost", port), port)
+	defer cls.Shutdown()
+	config := cls.DefaultConfig()
+	c, err := hz.StartNewClientWithConfig(ctx, config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer c.Shutdown(ctx)
+	db := driver.Open(config)
+	defer db.Close()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	rows, err := db.QueryContext(ctx, "select * from table(generate_stream(1))")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var v string
+		if err := rows.Scan(&v); err != nil {
+			t.Fatal(err)
+		}
+		if v == "2" {
+			cls.Shutdown()
+			go func() {
+				// check if it respects context cancel
+				<-time.After(2 * time.Second)
+				cancel()
+			}()
+		}
+		t.Log(v)
+	}
+	err = rows.Err()
+	require.Contains(t, err.Error(), "context canceled")
```

Comment:
Would it be better to check using `errors.Is`?:
```
if !errors.Is(err, context.Canceled) {
   t.Fatal("expected context canceled error")
}

Reply:
https://github.com/hazelcast/hazelcast-go-client/pull/862/commits/46d23408a379bab46104a824d03c0e1f8ab793e9